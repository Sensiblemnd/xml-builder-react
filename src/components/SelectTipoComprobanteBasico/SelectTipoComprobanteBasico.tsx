import React from "react";
import SelectCatalog from "../SelectCatalog";
import { SelectCatalogControlProps } from "../SelectCatalog/SelectCatalog";

export type OptionType =
  | ""
  | "FACTURA"
  | "BOLETA"
  | "NOTA_CREDITO"
  | "NOTA_DEBITO";

export const options: Map<OptionType, string> = new Map([
  ["", "Seleccione"],
  ["FACTURA", "FACTURA"],
  ["BOLETA", "BOLETA"],
  ["NOTA_CREDITO", "NOTA DE CRÉDITO"],
  ["NOTA_DEBITO", "NOTA DE DÉBITO"]
]);

class SelectTipoComprobanteBasico extends React.Component<
  SelectCatalogControlProps
> {
  render() {
    const { error, value, onChange, ...rest } = this.props;
    return (
      <SelectCatalog
        options={options}
        value={value}
        onChange={onChange}
        error={error}
        {...rest}
      />
    );
  }
}

export default SelectTipoComprobanteBasico;
