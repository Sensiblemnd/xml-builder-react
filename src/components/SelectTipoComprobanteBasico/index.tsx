import SelectTipoComprobanteBasico from "./SelectTipoComprobanteBasico";

export default SelectTipoComprobanteBasico;
