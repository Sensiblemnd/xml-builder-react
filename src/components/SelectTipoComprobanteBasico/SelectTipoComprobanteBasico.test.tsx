import React from "react";
import { shallow } from "enzyme";
import SelectTipoComprobanteBasico from "./SelectTipoComprobanteBasico";

it("Renders without crashing", () => {
  const wrapper = shallow(<SelectTipoComprobanteBasico />);

  expect(wrapper).toMatchSnapshot();
});
