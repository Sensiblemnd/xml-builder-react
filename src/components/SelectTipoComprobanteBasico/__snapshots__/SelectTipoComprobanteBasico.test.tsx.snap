// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders without crashing 1`] = `
<SelectCatalog
  options={
    Map {
      "" => "Seleccione",
      "FACTURA" => "FACTURA",
      "BOLETA" => "BOLETA",
      "NOTA_CREDITO" => "NOTA DE CRÉDITO",
      "NOTA_DEBITO" => "NOTA DE DÉBITO",
    }
  }
/>
`;
